common/plugin.hpp

```diff
@@ -63,6 +69,24 @@ class plugin {
         return name;
     }
 
+    void spdlogger(const char* log_level) {
```

Comment:
The plugins shouldn't have to define the log levels right? Wouldn't we need to read the log level from whatever configuration we're using?

Reply:
I think it might be good to overload this function to have one without any parameters. If called, it would query the plugin name, convert it to all caps, and try to read the env var for the log level.

Reply:
I look at it as they "get to" define the log levels. But to answer your question, that was why I changed it from a std::string to a const char *, so if the user doesn't define an env var then std::getenv() will return a NULL which can be checked for easily. Then based on one of Jeffrey's comments, an if statement will set it to either "warn" if compiling opt or "debug" if compiling dbg. But if the user did define an env var, then it is set to that level, regardless of how it's compiled. This seems like a decent compromise for now, but I can certainly see this as something to revisit once folks have had the time to use it.